src/coreclr/src/jit/emitarm64.cpp

```diff
@@ -4114,8 +4114,8 @@ void emitter::emitIns_R_R(
             else // ins == INS_xtn2
             {
                 assert(size == EA_16BYTE);
+                assert(elemsize != EA_8BYTE); // Narrowing must not end with 8 byte data
```

Comment:
Could you reword this comment?

It makes it sound like the result can't be `Vector128<byte>`, which is valid. Instead, the expectation is that the input isn't `Vector64<byte>`.